instrumentation/go.mongodb.org/mongo-driver/v2/mongo/otelmongo/mongo.go

```diff
@@ -0,0 +1,150 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelmongo // import "go.opentelemetry.io/contrib/instrumentation/go.mongodb.org/mongo-driver/v2/mongo/otelmongo"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
```

Comment:
Since this is a new package, we should rely on the latest semantic conventions ([1.30.0](https://github.com/open-telemetry/opentelemetry-go/tree/main/semconv/v1.30.0)) right away, let's not make things harder by forcing a migration right away :)

Reply:
there's some breaking changes between v1.21.0 and v1.30.0.  Most are simple name changes when comparing the semver key names in the docs, but `db.statement` was renamed in https://github.com/open-telemetry/opentelemetry-go/pull/5476.  I'll update it to `db.query.text`.

Reply:
Yes, that's exactly why we should rely on the latest. If we ship with the old version, we'll have to migrate in a forward PR.